src/cdk-experimental/menu/menu-item-trigger.spec.ts

```diff
@@ -38,20 +40,288 @@ describe('MenuItemTrigger', () => {
       expect(menuItemElement.getAttribute('aria-disabled')).toBe('true');
     });
 
+    it('should set aria-haspopup to menu', () => {
+      expect(menuItemElement.getAttribute('aria-haspopup')).toEqual('menu');
+    });
+
     it('should be a button type', () => {
       expect(menuItemElement.getAttribute('type')).toBe('button');
     });
 
     it('should  have a submenu', () => {
-      expect(menuItem.hasSubmenu).toBeTrue();
+      expect(menuItem.hasSubmenu()).toBeTrue();
+    });
+  });
+
+  describe('with nested sub-menus', () => {
+    let fixture: ComponentFixture<MenuBarWithNestedSubMenus>;
+
+    let menus: CdkMenu[];
+    let nativeMenus: HTMLElement[];
+    let menuItems: CdkMenuItem[];
+    let triggers: CdkMenuItemTrigger[];
+    let nativeTriggers: HTMLButtonElement[];
+
+    const setElements = () => {
+      menus = fixture.debugElement
+        .queryAll(By.directive(CdkMenu))
+        .map(e => e.injector.get(CdkMenu));
+      nativeMenus = fixture.debugElement.queryAll(By.directive(CdkMenu)).map(e => e.nativeElement);
+
+      menuItems = fixture.debugElement
+        .queryAll(By.directive(CdkMenuItem))
+        .map(e => e.injector.get(CdkMenuItem));
+      triggers = fixture.debugElement
+        .queryAll(By.directive(CdkMenuItemTrigger))
+        .map(e => e.injector.get(CdkMenuItemTrigger));
+      nativeTriggers = fixture.debugElement
+        .queryAll(By.directive(CdkMenuItemTrigger))
+        .map(e => e.nativeElement);
+    };
+
+    /** run change detection and, extract and set the rendered elements */
+    const detectChanges = () => {
+      fixture.detectChanges();
+      setElements();
+    };
+
+    const setDocumentDirection = (dir: 'ltr' | 'rtl') => (document.dir = dir);
+
+    beforeEach(async(() => {
+      TestBed.configureTestingModule({
+        imports: [CdkMenuModule],
+        declarations: [MenuBarWithNestedSubMenus],
+      }).compileComponents();
+    }));
+
+    beforeEach(() => {
+      fixture = TestBed.createComponent(MenuBarWithNestedSubMenus);
+      detectChanges();
+    });
+
+    afterEach(() => {
+      setDocumentDirection('ltr');
+    });
+
+    it('should toggle the aria-expanded attribute', () => {
+      expect(nativeTriggers[0].getAttribute('aria-expanded')).toBeNull();
+
+      triggers[0].toggle();
+      detectChanges();
+      expect(nativeTriggers[0].getAttribute('aria-expanded')).toEqual('true');
+
+      triggers[0].toggle();
+      detectChanges();
+      expect(nativeTriggers[0].getAttribute('aria-expanded')).toBeNull();
+    });
+
+    it('should hide sub-menus on initial load', () => {
+      expect(menus.length).toEqual(0);
+    });
+
+    it('should only open the attached submenu', () => {
+      triggers[0].toggle();
+      detectChanges();
+
+      expect(menus.length).toEqual(1);
+      expect(triggers[0]._menuPanel!._menu!).toEqual(menus[0]);
```

Comment:
I check that there is only one opened menu and that the opened menu is the one which is attached to the trigger.

Reply:
But since the 'menus' array is populated dynamically, is that enough to verify that the correct menu is opened? (If _menuPanel on the trigger is known to refer to the correct menu, _before_ the menu is opened, then that would do it.)

Reply:
ya, so the _menuPanel menu reference doesn't change since it's set in the template. Looking at it now, they should be flipped though - that's probably where the confusion is.